### DISCUSSION TITLE
Self hosted built-in presets

### DISCUSSION THREAD

**Discussion by ifalex** [renovatebot/renovate#22400] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.138.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Haven't seen anywhere a topic regarding self built-in presets.
From my understanding on reading and using shareable config presets, these shareable jsons are simply one/more rules combined in a typical custom configuration.
Our need would be to know how to create a shareable preset with parametrization.
For example:
1) Config preset for updating helm subcharts with filtering parameter for unwanted subcharts updates.
2) Update docker images in values.yaml file with filter parameter for unwanted images (might be cases when some of the images are updated via CI process)
As per my understanding built-in presets are .ts modules https://github.com/renovatebot/renovate/tree/main/lib/config/presets/internal
Can we have our own private presets (if yes, how?), or simply we need to contribute to github renovate?

### What you tried so far.

Used shareable config presets as specified in official documentation.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
You can have your own - you need to host them in a repository

**Reply by ifalex** [reply]
Can you elaborate a bit? A documentation on how to create your own parametrized preset?

**Reply by rarkins** [reply]
Have you read https://docs.renovatebot.com/config-presets/ to begin with?

It so then look at some of the parameterized presets in this repo source code and copy the concepts

**Reply by ifalex** [reply]
Yes, I have read it and actually tried to put in practice. My understanding from what I've read in the official doc lead me to this try:

preset-branches.json
`{
    "description": "Preset for branches",
    "baseBranches": ["arg0", "arg1"]
}
`
Trying to extend it in another renovate.json:

Using extend from a named, 'pathed' local repository (self-hosted gitlab)
`
local>org/repo//my-presets/preset-branches(main, /^release\\/.*/)
`
I try to extend and add parameters for the preset created.
This leads me to the following warn:

`
 WARN: Base branch does not exist - skipping (repository=org/repo, baseBranch=arg0)
 WARN: Base branch does not exist - skipping (repository=org/repo, baseBranch=arg1)
`

So how to use it properly?

**Reply by viceice** [reply]
the syntax of your placeholder args is wrong, please check them

**Reply by ifalex** [reply]
Yep, found it. Do you accept also lists as parameters?

**Reply by ifalex** [reply]
For later users that may find this, a detailed explanation on how can be proceeded:
Let's say you have a gitlab selfhosted with the following group: XUser and you plan to keep your shareable preset files into my-presets, you do have the option to use shareable config presets from a path:
The repo will look then like this:
XUser/my-presets
|_ my-renovate-shareable-presets
   |_ preset-branches.json

If you plan to parametrize this preset then you should do the following:
- add as many arguments as you want: "{{arg0}}", "{{arg1}}" etc

`
{
   "baseBranches": ["{{arg0}}", "{{arg1}}"]
}
`

To call it from another repo with parameters simply add to your renovate.json the following 

renovate.json
`
{
"extends": [ "local>XUser/my-presets//my-renovate-shareable-presets/preset-branches(main, branch1)
}
`
The example below refers to baseBranches which accept a list as value but you can accommodate it to every configuration from renovate.

**Reply by ifalex** [reply]
Last question: can arguments be parametrized as lists? Or simply strings?

**Reply by rarkins** [reply]
I just noticed that you've been giving thumbs-down responses to our help. Take a break

**Reply by ifalex** [reply]
Something like this:
`local>.../preset-branches([main,mynotbranch])` instead of `local>.../preset-branches(main,mynotbranch)`
